internal/pkg/ai-functions/functions/test-case/function_test.go

```diff
@@ -0,0 +1,185 @@
+// Copyright (c) 2021 Terminus, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package test_case
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"testing"
+
+	"bou.ke/monkey"
+
+	"github.com/erda-project/erda-proto-go/apps/aifunction/pb"
+	"github.com/erda-project/erda/apistructs"
+	"github.com/erda-project/erda/bundle"
+)
+
+func TestFunction_Callback(t *testing.T) {
+	type fields struct {
+		background *pb.Background
+	}
+	type args struct {
+		ctx        context.Context
+		arguments  json.RawMessage
+		input      interface{}
+		needAdjust bool
+	}
+	input := TestCaseFunctionInput{
+		TestSetID: 24227,
+		IssueID:   30001127564,
+		Prompt:    "用户登录",
+	}
+
+	results := []apistructs.TestCaseStepAndResult{
+		{
+			Result: "登录成功，跳转到用户个人主页",
+			Step:   "用户输入有效的用户名和密码",
+		},
+		{
+			Result: "下次登录自动填充用户名和密码",
+			Step:   "用户输入记住密码",
+		},
+		{
+			Result: "跳转到密码重置页面",
+			Step:   "用户点击忘记密码",
+		},
+		{
+			Result: "跳转到用户注册页面",
+			Step:   "用户点击注册",
+		},
+	}
+
+	arguments := []byte("{\"name\":\"用户登录\",\"preCondition\":\"用户打开登录页面\",\"stepAndResults\":[{\"step\":\"用户输入有效的用户名和密码\",\"result\":\"登录成功，跳转到用户个人主页\"},{\"step\":\"用户输入记住密码\",\"result\":\"下次登录自动填充用户名和密码\"},{\"step\":\"用户点击忘记密码\",\"result\":\"跳转到密码重置页面\"},{\"step\":\"用户点击注册\",\"result\":\"跳转到用户注册页面\"}]}")
```

Comment:
use struct and then marshal it as arguments.

Reply:
OK